src/DistributedMetadata/CatalogService.cpp

```diff
@@ -576,13 +605,26 @@ CatalogService::TableContainerPerNode CatalogService::buildCatalog(const NodePtr
 /// Merge a snapshot of tables from one host
 void CatalogService::mergeCatalog(const NodePtr & node, TableContainerPerNode snapshot)
 {
+    std::unique_lock guard{catalog_rwlock};
+
     if (snapshot.empty())
-    {
+    {   
+        /// Clear the table corresponding residual data
```

Comment:
After refactoring out, we don't need this comment anymore coz the function name already tells what it does

Reply:
Removed